### PR TITLE
Sim running

### PR DESCRIPTION
**Pull request by tashoecraft**
The sim is wired up to work. You first connect, and that will connect to the simulated board and begin transmitting data. If you want to see the stream click, "start stream" and it will log each incoming object. Clicking "stop stream" won't actually stop the stream, merely stop logging.

### CODE REVIEW

**Comment by andrewjaykeller** [comment]
Shouldn't .DS_Store be on the gitignore?

File: client/components/board/boardFactory.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                .then(boardSerial => {
+                    ourBoard.on('ready',function() {
+                        /** Start streaming, reading registers, what ever your heart desires  */
+                    });
+                }).catch(function(err) {
+                /** Handle connection errors */
+            });
+        };
+
+        var observer = function(changes) {
```
Comment (andrewheusser) [inline client/components/board/boardFactory.js:30]: is there a reason you use the function() syntax instead of the => ?  Just trying to educate myself :)

File: client/components/board/simulatorFactory.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        const OpenBCIBoard = require('openbci-sdk');
+        const ourBoard = new OpenBCIBoard.OpenBCIBoard();
+
+        var storage = {};
+
+        var simulatorStart = () => {
+            ourBoard.simulatorStart()
+                .then(() => {
+                    ourBoard.on('sample',sample => {
+                        storage[sample.sampleNumber] = sample;
```
Comment (andrewheusser) [inline client/components/board/simulatorFactory.js:12]: does this mean that all samples will be saved in memory?  we may have to deal with this down the line..

File: client/app/connect/connect.controller.js [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
+
+        $scope.simStart = simulatorFactory.simulatorStart;
+
+        $scope.startStream = simulatorFactory.publish;
+        $scope.stopStream = simulatorFactory.unpublish;
+
+        $scope.listDevices = () => {
+            boardFactory.findPorts()
+                .then(ports => {
+                    $scope.$apply(_.merge($scope.devices, ports));
```
Comment (andrewheusser) [inline client/app/connect/connect.controller.js:19]: am i right to say that $apply is the manual way to update the digest loop?  why do you have to do that here?

File: client/components/board/simulatorFactory.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                    })
+                }).catch(err => console.log('Error [simulator]: ' + err))
+        };
+
+        var observer = function(changes) {
```
Comment (andrewheusser) [inline client/components/board/simulatorFactory.js:17]: I'm confused about this code, can you summarize what's happening here?

**Comment by andrewheusser** [comment]
LGTM @tashoecraft , will merge but can you still comment?  Thanks!

File: client/components/board/boardFactory.js (hunk 1 above)
Comment (tashoecraft) [inline client/components/board/boardFactory.js:30]: That's because  I copied it from @pushtheworldllc Readme for the sdk and didn't change it as it isn't hooked up.

File: client/components/board/simulatorFactory.js (hunk 2 above)
Comment (tashoecraft) [inline client/components/board/simulatorFactory.js:12]: currently yes, but that can be changed by having it delete some, or store them in a database when it reaches a certain age.

File: client/app/connect/connect.controller.js (hunk 3 above)
Comment (tashoecraft) [inline client/app/connect/connect.controller.js:19]: You're correct. It was a bug that I didn't expect, but doing the merge Angular wasn't firing it's watcher on devices. So I wrapped it in apply and all is well.

File: client/components/board/simulatorFactory.js (hunk 4 above)
Comment (tashoecraft) [inline client/components/board/simulatorFactory.js:17]: we turn on the simulator with simulatorStart(), then once we receive a 'sample', which means the simulator is started, we add to the storage object the value of sample. So inside storage are a bunch of keys that are the sampleNumbers, 1,2,3,4... whose value is the entire sample. Just a easy way of storing. The observer is essentially a watcher that just dirty checks storage to see if anything has changed, if it has then it fires
